realm/realm-library/src/androidTest/java/io/realm/rule/RunInLooperThread.java

```diff
@@ -83,13 +82,11 @@ public void run() {
                             backgroundLooper[0] = Looper.myLooper();
                             try {
                                 realm = Realm.getInstance(realmConfiguration);
-
                                 base.evaluate();
-
                                 Looper.loop();
                             } catch (Throwable e) {
                                 threadAssertionError[0] = e;
-
+                                unitTestFailed = true;
```

Comment:
I am not sure if `threadAssertionError[0]` is still necessary. JUNI4 will just stop when any thread throws.